doc/getting-started.md

```diff
@@ -0,0 +1,584 @@
+    match token {
+        SERVER => {
+            // Only receive readable events
+            assert!(events.is_readable());
+
+            println!("the server socket is ready to accept a connection");
+            match self.server.accept() {
+                Ok(Some(connection)) => {
+                    println!("accepted a socket, exiting program");
+                    event_loop.shutdown();
+                }
+                Ok(None) => {
+                    println!("the server socket wasn't actually ready");
+                }
+                Err(e) => {
+                    println!("listener.accept() errored: {}", e);
+                    event_loop.shutdown();
+                }
+            }
+        }
+        _ => panic!("Received unknown token"),
+    }
+}
+```
+
+Since we only ever registered one socket with the event loop, the
+`ready` handler will only ever be called with the `SERVER` token.
+Then, we try to accept a connection. It's important to note that the
+event loop will never operate on any sockets. It only watches sockets
+for changes in state.
+
+The signature for the `TcpListener::accept()` function is as follows:
+
+```no_run
+fn accept(&self) -> io::Result<Option<TcpStream>>;
+```
+
+All non-blocking socket types follow a similar pattern of returning
+`io::Result<Option<T>>`. An operation on a non-blocking socket can
+obviously flat out fail and return an error. However, the socket can
+also be in a good state, but not be ready to operate on. If it were a
+blocking socket, the operation would block. Instead, it returns
+immediately with Ok(None), in which case we must wait for the event loop
+to notify us again that the socket is readable.
+
+> **Important:** Even if we just received a ready notification, there is
+> no guarantee that a read from the socket will succeed and not return
+> `Ok(None)`, so we must handle that case as well.
+
+If a connection was successfully accepted, we just print some output and
+shutdown the event loop. The `event_loop.run(...)` call will return and
+the program will exit.
+
+### Handling connections
+
+Our echo server will be pretty simple. It will receive data off of the
+connection, buffer it until it sees a newline, and then return the data
+to the client.
+
+First, we need a way to store the connection state on the `Pong` server.
+To do this, we are going to create a `Connection` struct that will
+contain the state for a single client connection. The `Pong` struct will
+use a `Slab` to store all the `Connection` instances.
+
+A `Slab` is a fixed capacity map of `Token`, the type that mio uses to
+identify sockets (discussed below) to T, defined by the user. In this
+case, it will be a `Slab<Connection>`. The advantage of using a slab
+over a HashMap is that it is much lighter weight and optimized for use
+with mio.
+
+Our `Pong` struct looks like this now:
+
+```rust
+struct Pong {
+    server: TcpListener,
+    connections: Slab<Connection>,
+}
+```
+
+#### Tokens
```

Comment:
some of my token comments could be mitigated by moving this up higher, as is I would just be confused for quite some time.